### PR TITLE
Add configuration to access gitleaks secrets from test-infra

### PR DESCRIPTION
**Pull request by KacperMalachowski**
**Description**

Changes proposed in this pull request:

- Add service account to access the gitleaks secrets
- Add binding to workload identity federation
- Add configfor github variables required by workload identity federation 

**Related issue(s)**

/kind feature
/area security

### CODE REVIEW

File: configs/terraform/environments/prod/gitleaks.tf [hunk 1]
```diff
@@ -6,33 +6,24 @@ resource "google_service_account" "gitleaks_secret_accesor" {
...
   role               = "roles/iam.workloadIdentityUser"
   service_account_id = google_service_account.gitleaks_secret_accesor.name
 }
 
-resource "github_actions_variable" "gcp_gitleaks_sercet_accesor_service_account_email" {
-  for_each      = data.github_repository.gitleaks_repository
+resource "github_actions_organization_variable" "github_gitleaks_secret_accesot_service_account_email" {
   provider      = github.kyma_project
-  repository    = each.key
   variable_name = "GCP_GITLEAKS_SERCET_ACCESOR_SERVICE_ACCOUNT_EMAIL"
```
Comment (dekiel) [inline configs/terraform/environments/prod/gitleaks.tf:19]:
```suggestion
  variable_name = "GCP_GITLEAKS_SECRET_ACCESOR_SERVICE_ACCOUNT_EMAIL"
```

File: configs/terraform/environments/prod/gitleaks.tf [hunk 2]
```diff
@@ -6,33 +6,24 @@ resource "google_service_account" "gitleaks_secret_accesor" {
...
-  for_each = var.gitleaks_repositories
-  provider = github.kyma_project
-  name     = each.value
-}
-
 resource "google_service_account_iam_binding" "gitleaks_workload_identity_federation_binding" {
-  for_each = data.github_repository.gitleaks_repository
   members = [
-    "principal://iam.googleapis.com/projects/${data.google_client_config.gcp.project}/locations/global/workloadIdentityPools/${module.gh_com_kyma_project_workload_identity_federation.pool_name}/subject/repository_id:${each.value.repo_id}:repository_owner_id:${var.github_kyma_project_organization_id}:workflow:${var.gitleaks_workflow_name}"
+    "principal://iam.googleapis.com/projects/${data.google_client_config.gcp.project}/locations/global/workloadIdentityPools/${module.gh_com_kyma_project_workload_identity_federation.pool_name}/subject/repository_owner_id:${var.github_kyma_project_organization_id}:workflow:${var.gitleaks_workflow_name}"
```
Comment (dekiel) [inline configs/terraform/environments/prod/gitleaks.tf:11]: Can we use data for kyma-project organization id instead using variable?

**Review by dekiel — CHANGES_REQUESTED** [review]
Please extend comment for gitleaks.tf and gitleaks-variables.tf. I would add some details that we need this github variables because we use them in gh workflow. That we allow to use this service account from workflows in all repos in our organisation.

Think about information which are now obvious, because we actively work on the topic but which might be useful for someone new in the topic, especially after some time.

File: configs/terraform/environments/prod/gitleaks.tf (hunk 2 above)
Comment (KacperMalachowski) [reply configs/terraform/environments/prod/gitleaks.tf:11]: The data source for github organizaiton doesn't provide such information. We have to stick with variable until we found different provider

**Pull request merged**